py/server/deephaven/table.py

```diff
@@ -462,40 +522,40 @@ def size(self) -> int:
         """The current number of rows in the table."""
         return self.j_table.size()
 
-    @property
+    @cached_property
     def is_refreshing(self) -> bool:
         """Whether this table is refreshing."""
-        if self._is_refreshing is None:
-            self._is_refreshing = self.j_table.isRefreshing()
-        return self._is_refreshing
+        return self.j_table.isRefreshing()
 
-    @property
+    @cached_property
     def is_blink(self) -> bool:
         """Whether this table is a blink table."""
         return _JBlinkTableTools.isBlink(self.j_table)
 
-    @property
+    @cached_property
     def update_graph(self) -> UpdateGraph:
         """The update graph of the table."""
-        if self._update_graph is None:
-            self._update_graph = UpdateGraph(self.j_table.getUpdateGraph())
-        return self._update_graph
+        return UpdateGraph(self.j_table.getUpdateGraph())
 
-    @property
+    @cached_property
     def is_flat(self) -> bool:
         """Whether this table is guaranteed to be flat, i.e. its row set will be from 0 to number of rows - 1."""
-        if self._is_flat is None:
-            self._is_flat = self.j_table.isFlat()
-        return self._is_flat
+        return self.j_table.isFlat()
 
     @property
-    def columns(self) -> List[Column]:
-        """The column definitions of the table."""
-        if self._schema:
-            return self._schema
+    def definition(self) -> TableDefinition:
+        """The table definition."""
+        return self._definition
 
-        self._schema = _td_to_columns(self._definition)
-        return self._schema
+    @property
```

Comment:
cache?

Reply:
The primary reason to cache is to prevent jpy java calls. In all the cases where cache_property isn't used, it's either all pure python, or the delegated layer is handling the caching.